src/patient-lab-dashboard/upload-report/upload-report.resources.tsx

```diff
@@ -138,3 +147,69 @@ const removeBase64 = fileData => {
     fileData.length,
   )
 }
+
+export const bahmniEncounter = (
+  locationUuid: string,
+  patientUuid: string,
+  providerUuid: string,
+  selectedTest: LabTest,
+  selectedPendingOrder: PendingLabOrders[],
+  ac: AbortController,
+) => {
+  let requestBody: BahmniEncounteRequestType
+  if (providerUuid && providerUuid !== null) {
+    requestBody = {
+      locationUuid,
+      patientUuid,
+      encounterUuid: null,
+      visitUuid: null,
+      providers: [
+        {
+          uuid: providerUuid,
+        },
+      ],
+      visitType: 'OPD',
+      encounterTypeUuid: getEncounterTypeUuid(
+        selectedTest,
+        selectedPendingOrder,
+      ),
+    }
+  } else {
+    requestBody = {
+      locationUuid,
+      patientUuid,
+      encounterUuid: null,
+      visitUuid: null,
+      visitType: 'OPD',
+      encounterTypeUuid: getEncounterTypeUuid(
+        selectedTest,
+        selectedPendingOrder,
+      ),
+    }
+  }
+  return postApiCall(bahmniEncounterUrl, requestBody, ac)
```

Comment:
is this handling API failure scenario also ?

Reply:
incorporated the feedback